insights/client/crypto.py

```diff
@@ -0,0 +1,205 @@
+import errno
+import os.path
+import logging
+import shutil
+import tempfile
+import subprocess
+
+logger = logging.getLogger(__file__)
+
+
+class CommandResult(object):
+    """Output of a command.
+
+    Attributes:
+        ok (bool): Result of an operation.
+        return_code (int): Return code of the operation.
+        stdout (str): Standard output of the command.
+        stderr (str): Standard error of the command.
+        _command: An optional reference to the object that created the result
+    """
+    def __init__(self, ok, return_code, stdout, stderr):
+        self.ok = ok
+        self.return_code = return_code
+        self.stdout = stdout
+        self.stderr = stderr
+        self._command = None
```

Comment:
This attribute that is not set by a constructor argument, but rather by an assignment on an existing object is something that a [namedtuple](https://docs.python.org/2.6/library/collections.html#collections.namedtuple) can’t do. A [dataclass](https://docs.python.org/3/library/dataclasses.html) would, but we can’t use it. I’d rarely see mutating existing objects as a good practice, and doing so only not to have an ommitable value doesn’t seem like something that’d deserve an exception.

Reply:
I wouldn't call this a mutation, I understand it as an extended context of the command. It is a private attribute, which implicitly assumes you know what you are doing.
But sure, I've altered the initializer to take a command argument as well.